src/test/java/com/LoginTest.java

```diff
@@ -11,9 +13,39 @@ public class LoginTest extends BasicTest {
     @Test()
     public void loginTest() throws Exception {
         // Launch website
-        String url = "https://automationtestpro.com/";
+        String url = "https://bantheme.xyz/hathanhauto/tai-khoan/";
         driver.get(url);
         Assert.assertEquals(driver.getCurrentUrl(), url);
+        
+        //Step1: Click  submit button
+        WebElement submitBtn = driver.findElement(By.xpath("//button[contains(text(),'Đăng ký')]"));
+        submitBtn.click();
+        WebElement errorMessage = driver.findElement(By.xpath("//div[@id='main']//li[1]"));
+        System.out.println(errorMessage.getText());
+        Assert.assertEquals(errorMessage.getText(),"Lỗi: Vui lòng cung cấp địa chỉ email hợp lệ.");
+
+        //Step2: Email invalid
+        WebElement email = driver.findElement(By.xpath("//input[@id='reg_email']"));
+        email.sendKeys("123@456");
+        WebElement submitBtn2 = driver.findElement(By.xpath("//button[contains(text(),'Đăng ký')]"));
+        submitBtn2.click();
+        WebElement errorMessage2 = driver.findElement(By.xpath("//div[@id='main']//li[1]"));
+        Assert.assertEquals(errorMessage2.getText(),"Lỗi: Vui lòng cung cấp địa chỉ email hợp lệ.");
+
+
+        //Step3:Valid email, missing password
+        WebElement email3 = driver.findElement(By.xpath("//input[@id='reg_email']"));
+        email3.clear();
+        email3.sendKeys("thuytran@gmail.com");
+        WebElement submitBtn3 = driver.findElement(By.xpath("//button[contains(text(),'Đăng ký')]"));
+        submitBtn3.click();
+        WebElement errorMessage3 = driver.findElement(By.xpath("//div[@id='main']//li[1]"));
+        Assert.assertEquals(errorMessage3.getText(),"Lỗi: Vui lòng nhập mật khẩu tài khoản.");
+
+        //QUESTION: khong biet sao mỗi lần e lại phải khai báo lại biến email, buttonb, error nếu không chạy sẽ báo lỗi không tìm thấy phần tủ.tủ
```

Comment:
Do cai web minh practise no' cui` qua' em, moi lan` click submit la no reload lai web nen no de~ bi loi~ stale elemen exception. 
De fix no thi minh hoi phuc tap. xiu, em phai wait va` catch no toi khi no xuat hien.
Ma khi lam cai proj thuc te e se kho' gap issue nay

Reply:
sample code:

```
/**
     * Attempts to click on an element multiple times (to avoid stale element
     * exceptions caused by rapid DOM refreshes)
     *
     * @param d
     *            The WebDriver
     * @param by
     *            By element locator
     */
    public static void dependableClick(WebDriver d, By by)
    {
        final int MAXIMUM_WAIT_TIME = 10;
        final int MAX_STALE_ELEMENT_RETRIES = 5;

        WebDriverWait wait = new WebDriverWait(d, MAXIMUM_WAIT_TIME);
        int retries = 0;
        while (true)
        {
            try
            {
                wait.until(ExpectedConditions.elementToBeClickable(by)).click();

                return;
            }
            catch (StaleElementReferenceException e)
            {
                if (retries < MAX_STALE_ELEMENT_RETRIES)
                {
                    retries++;
                    continue;
                }
                else
                {
                    throw e;
                }
            }
        }
    }
```